ui/packages/consul-ui/app/services/repository/hcp-link.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import RepositoryService from 'consul-ui/services/repository';
+import dataSource from 'consul-ui/decorators/data-source';
+
+export default class HcpLinkService extends RepositoryService {
+  @dataSource('/:partition/:ns/:dc/hcp-link')
+  async fetch({ partition, ns, dc }, { uri }, request) {
+    let result;
+    try {
+      result = (
+        await request`
+      GET /api/hcp/v2/link/global
+    `
+      )((headers, body) => {
+        return {
+          meta: {
+            version: 2,
+            uri: uri,
+          },
+          body: {
+            isLinked: (body.status['consul.io/hcp/link']['conditions'] || []).some(
+              (condition) => condition.type === 'linked' && condition.state === 'STATE_TRUE'
```

Comment:
Was going to ask why we have the array methods here, but the API response does have `conditions` to be an array, I wonder if we'd ever see multiple contradicting conditions?

I hope not - feels like something we should handle in the backend

Reply:
yea, it is how the backend responds as of now. I've just decided to check that condition is linked, just in case we will have some other stuff added in future. 

But I totally agree, that as of now we don't assume to have something else to be returned from the endpoint